### PR TITLE
Add Gluon fused MoE support for dynamic MXFP4 activations

### PR DESCRIPTION
**Pull request by qedawkins**
## Summary

- Add gfx950 Gluon MXFP4 MoE for dynamic MXFP4 activations.
- Add BF16/FP16 -> MXFP4 activation quantization between expert GEMMs with swizzled/padded ragged scales.
- Add fused small-M decode routing/top-k metadata for MXFP4 MoE, falling back when `M * topk > 64`.
- Handle Kimi W13 gate/up layout, SiLU gate/up, and `norm_topk_prob` semantics.

## Testing Plan

This PR adds gfx950 coverage for:

- Fused dynamic MXFP4 MoE decode vs torch reference.
- Kimi concatenated W13 preprocessing + SiLU gate/up correctness.
- Routed/pregathered dynamic MXFP4 activation scales.
- Large expert-offset addressing.
- W13 gate/up interleave conversion.
- Grouped-routing config validation.
- Kernel API selection for `gluon_mxfp4_dynamic_moe_apply`.

### Synthetic TP4 MoE apply

AMD Instinct MI355X. Kimi TP4 dynamic MXFP4 MoE: `H=7168`, `I/rank=512`, `E=384`, `topk=8`, `n_group=1`, `topk_group=1`. Warmup/repeat: `100/1000`. Outputs checked with `rtol=3e-2`, `atol=3e-3`.

Fused route is active for `M * topk <= 64`; with Kimi `topk=8`, that means `tokens <= 8`.

| Tokens | Routed rows | Unfused median | Fused median | Speedup |
| ---: | ---: | ---: | ---: | ---: |
| 1 | 8 | 0.523 ms | 0.235 ms | 2.23x |
| 4 | 32 | 0.519 ms | 0.239 ms | 2.17x |
| 8 | 64 | 0.534 ms | 0.255 ms | 2.09x |

### Kimi TP4 e2e smoke sweep

`amd/Kimi-K2.5-MXFP4`, TP4, EvalScope random, 512 input / 32 output tokens, no warmup.

| Concurrency | Requests | Success | Gen/s | Total tok/s | Avg TTFT | Avg TPOT | Decode tok/s |
| ---: | ---: | ---: | ---: | ---: | ---: | ---: | ---: |
| 1 | 1 | 100% | 13.91 | 236.47 | 1508.8 ms | 25.5 ms | 39.15 |
| 2 | 4 | 100% | 30.40 | 516.83 | 1266.8 ms | 26.8 ms | 37.30 |

Still overall quite slow but improving.

### CODE REVIEW

**Review by panditsa — APPROVED** [review]

**Comment by minminsun** [comment]
Is the precision ok with dynamic MXFP4 activations?

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/gluon/mxfp4.py [hunk 1]
```diff
@@ -33,14 +33,100 @@
...
+    def _swiglu_args(w: torch.nn.Module) -> tuple[float, float, float]:
+        swiglu_arg = getattr(w, "swiglu_arg", None)
+        if swiglu_arg is None:
+            return 1.0, 0.0, 0.0
+        swiglu_beta = getattr(w, "swiglu_beta", None)
+        return (
+            1.0 if swiglu_arg.alpha is None else swiglu_arg.alpha,
+            0.0 if swiglu_arg.limit is None else swiglu_arg.limit,
+            0.0 if swiglu_beta is None else swiglu_beta,
+        )
```
Comment (raikonenfnu) [inline tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/gluon/mxfp4.py:58]: QQ, does the 1.0, 0.0, 0.0 effectively what makes the swiglu -> silu? If so, thoughts on adding a little comment here? :)

**Comment by qedawkins** [comment]
> Is the precision ok with dynamic MXFP4 activations?

This is matching vllm's quant config for deepseekv3 style models: https://github.com/vllm-project/vllm/blob/79ca54d2215b22d9a4fc17378eb7aa2b2eb9dbd1/vllm/model_executor/layers/quantization/quark/quark.py#L49-L100

So precision should be fine and aime testing has been fine so far too.

File: tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/moe/mxfp4_gfx950_preprocess.py [hunk 2]
```diff
@@ -170,8 +185,7 @@ def _attach_gluon_preshuffle(w: torch.nn.Module) -> None:
             shuffled.original_n = int(logical_n)
             raw.original_n = int(logical_n)
             wrapped.original_n = int(logical_n)
-        raw._gluon_shuffled = shuffled
-        wrapped._gluon_shuffled = shuffled
+        setattr(w, attr, shuffled)
```
[the remark this answers is not in the document]
Comment (qedawkins) [reply tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/moe/mxfp4_gfx950_preprocess.py:188]: The issue is that we don't have enough memory to support carrying both tensors. We probably need to shift medium_decode  paths to use the shuffled layout instead, but no matter what replicating weights is not an acceptable solution in any case.

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/gluon/mxfp4.py (hunk 1 above)
Comment (qedawkins) [reply tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/gluon/mxfp4.py:58]: Done, added a quick comment
